lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -0,0 +1,436 @@
+                    self.log('Creating vault {0}'.format(self.name))
+                    use_sku = Sku(self.sku)
+                    permission_preset = create_permissions(self)
+                    access_policy_lst = [AccessPolicyEntry(self.tenant_id, self.object_id, permission_preset, application_id=self.application_id)]
+                    vault_properties = VaultProperties(self.tenant_id,
+                                                       use_sku,
+                                                       access_policies=access_policy_lst,
+                                                       vault_uri=self.vault_uri,
+                                                       enabled_for_deployment=self.enabled_for_deployment,
+                                                       enabled_for_disk_encryption=self.enabled_for_disk_encryption,
+                                                       enabled_for_template_deployment=self.enabled_for_template_deployment)
+                    vault = VaultCreateOrUpdateParameters(self.location, vault_properties, tags=self.tags)
+                else:
+                    # update the vault
+                    vault = VaultCreateOrUpdateParameters(self.location, results['properties'], tags=results['tags'])
+
+                self.results['state'] = self.create_or_update_vault(vault)
+            elif self.state == 'absent':
+                # delete vault
+                self.delete_vault()
+                # the delete does not actually return anything. if no exception, then we'll assume
+                # it worked.
+                self.results['state']['status'] = 'Deleted'
+        return self.results
+
+    def create_or_update_vault(self, vault):
+        try:
+            # create or update the new Vault object we created
+            new_vault = self.keyvault_client.vaults.create_or_update(self.resource_group, self.name, vault)
+        except Exception as exc:
+            self.fail("Error creating or updating vault {0} - {1}".format(self.name, str(exc)))
+        return new_vault.as_dict()
+
+    def delete_vault(self):
+        try:
+            # delete the Vault
+            self.keyvault_client.vaults.delete(self.resource_group, self.name)
+        except Exception as exc:
+            self.fail("Error deleting vault {0} - {1}".format(self.name, str(exc)))
+        return None
+
+
+def create_permissions(self):
+    # function takes in presets for permissions and returns a permissions object with the necessary/required permissions
+    fin_permissions = None
+    k_perm = []
+    s_perm = []
+    c_perm = []
+    # if they want the 'Key, Secret, & Certificate Management' permission preset, create and return a new permissions class with all the necessary information
+    if self.permissions == 'Key, Secret, & Certificate Management':
+        k_perm = ['Get', 'List', 'Update', 'Create', 'Import',
+                  'Delete', 'Recover', 'Backup', 'Restore']
```

Comment:
use ` KeyPermissions, SecretPermissions, StoragePermissions, CertificatePermissions` model in `key_vault_management_client_enums` intead of str

Reply:
done